crates/matrix-sdk-ui/src/sync_service.rs

```diff
@@ -219,14 +224,9 @@ impl SyncTaskSupervisor {
             }
         };
 
-        if let Err(err) = sender
-            .send(TerminationReport {
-                is_error,
-                has_expired,
-                origin: TerminationOrigin::EncryptionSync,
-            })
-            .await
-        {
+        let report =
+            TerminationReport { is_error, has_expired, origin: TerminationOrigin::EncryptionSync };
```

Comment:
I don't see the point of the `report` variables: they only add one level of "indirection" (some extra thing to keep in mind for a reader), they create one extra binding for the compiler frontend to handle, and they're used only once, so might as well be kept inlined at use.

Reply:
I prefer the formatting of the two statements compared to the formatting of the single statement, but ok: b38e8b94b55800bfa290b270107096f3a15b95f0.